mscore/prefsdialog.cpp

```diff
@@ -1100,15 +1113,28 @@ void PreferenceDialog::apply()
             mscore->update();
             }
 
+      QString partText = partStyle->text();
+      QString partPref = preferences.getString(PREF_SCORE_STYLE_PARTSTYLEFILE);
       if (defaultStyle->text() != preferences.getString(PREF_SCORE_STYLE_DEFAULTSTYLEFILE)) {
-            preferences.setPreference(PREF_SCORE_STYLE_DEFAULTSTYLEFILE, defaultStyle->text());
-            MScore::readDefaultStyle(preferences.getString(PREF_SCORE_STYLE_DEFAULTSTYLEFILE));
+            QString qs = defaultStyle->text();
+            preferences.setPreference(PREF_SCORE_STYLE_DEFAULTSTYLEFILE, qs);
+            if (qs.isEmpty()) {
+                  MScore::setDefaultStyle(MScore::baseStyle());
+                  if (partText.isEmpty() && partPref.isEmpty())
+                        MScore::setDefaultStyleForParts(new MStyle(MScore::defaultStyle()));
+                  }
+            else 
+                  MScore::readDefaultStyle(qs);
             }
 
-      if (partStyle->text() != preferences.getString(PREF_SCORE_STYLE_PARTSTYLEFILE)) {
-            preferences.setPreference(PREF_SCORE_STYLE_PARTSTYLEFILE, partStyle->text());
-            MScore::defaultStyleForPartsHasChanged();
-            }
+      if (partText != partPref) {
+            preferences.setPreference(PREF_SCORE_STYLE_PARTSTYLEFILE, partText);
+///!!!      MScore::defaultStyleForPartsHasChanged(); ///!!!it's an empty function, this is the only place it's called
+            if (partText.isEmpty())
+                  MScore::setDefaultStyleForParts(new MStyle(MScore::defaultStyle()));
+            else
+                  MScore::readDefaultStyle(partText);
+      }
```

Comment:
This brace should be indented one more level.

Reply:
check